pydiscourse/client.py

```diff
@@ -551,6 +551,19 @@ def delete_topic(self, topic_id, **kwargs):
         """
         return self._delete(u"/t/{0}".format(topic_id), **kwargs)
 
+    def watch_topic(self, topic_id, username, **kwargs):
+        """
+
+        Args:
+            topic_id:
+            **kwargs:
+
+        Returns:
+
+        """
+        self.api_username = username
```

Comment:
Why is this resetting the `api_username`? That's the username to connect to the API with, and it's already set when creating the client instance. Changing it here would presumably result in authentication errors trying to make requests to the Discourse API...